documentation/sphinx/source/disk-snapshot-backup.rst

```diff
@@ -0,0 +1,91 @@
+
+Disk snapshot backup and restore are generally high performant because it deals at disk level and data is not read or written through FoundationDB stack. In environments where disk snapshot and restore are highly performant this approach can be very fast. Feature is strictly dependent on crash consistent snapshot feature from disk system. Frequent backups could be done as a substitute to the continuos backup if the backups are performant.
+
+Limitations
+===========
+
+ * feature is not supported on Windows operating system
+ * data encryption is dependent on the disk system
+ * backup and resotre involves tooling which are deployment and environment specific to be developed by operators
+
+Backup Steps
+=============
+
+``snapshot``
+    This command line tool is used to create the snapshot. It takes a full path to a snapshot create binary and reports the status, optionally, can take additional arguments to be passed down to the snapshot create binary. It returns a Unique Identifier which can be used to identify all the disk snapshots of a backup.
+
+In response to the snapshot request from the user, FoundationDB will run a user specificed snapshot create binary on all processes which has persistent data in it, binary should call filesystem/disk system specific snapshot create API and gather some additional data for the restore.
+
+Before using the ``snapshot`` command the following setup needs to be done
+     * write a program that will take snapshot of the local disk store when invoked by the ``fdbserver`` with the following arugments:
+           * UID - 32 byte alpha-numeric unique identifier, the same identifier will be passed to all the nodes in the cluster, can be used to identify the set of disk snapshots associated with this backup
+           * Version - version string of the FoundationDB binary
+           * Path - path of the FoundationDB disk store to be snapshotted
+           * Role - tlog/storage/coordinator, identifies the role of the node on which the snapshot is being invoked
+     * install snapshot create binary on the FoundationDB instance in a secure path that can be invoked by the ``fdbserver``
+     * set a new config parameter ``whitelist_binpath`` for fdbserver section, whose value is the absolute snapshot create binary path. Running any snapshot command will validate that it is in the ``whitelist_binpath``. This is a security mechanism to stop running a random/unsecure command on the cluster by a client using ``snapshot`` command
+     * snapshot create program should capture any additional data needed to restore the cluster, additional data could be stored as tags in cloud environments or it could be stored in an additional file/directory in the data repository and then snapshotted. The section below describes a recommended specification of the list of things that can be gathered by snapshot create binary as part of backup to aid with restore.
```

Comment:
I'm assuming `snapshot create program` is the command to run. Currently none of the commands are wrapped inside inline code, and hence a bit harder to differentiate from rest of description. Hence ->

snapshot create program -> `snapshot create program`